.tekton/collector-pull-request.yaml

```diff
@@ -0,0 +1,391 @@
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+
+metadata:
+  annotations:
+    build.appstudio.openshift.io/repo: https://github.com/stackrox/collector?rev={{revision}}
+    build.appstudio.redhat.com/commit_sha: '{{revision}}'
+    build.appstudio.redhat.com/pull_request_number: '{{pull_request_number}}'
+    build.appstudio.redhat.com/target_branch: '{{target_branch}}'
+    pipelinesascode.tekton.dev/max-keep-runs: "500"
+    # TODO(ROX-21073): re-enable for all PR branches
+    pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && (source_branch.contains("rhtap") || source_branch.contains("konflux"))
+  creationTimestamp: null
+  labels:
+    appstudio.openshift.io/application: acs
+    appstudio.openshift.io/component: collector
+    pipelines.appstudio.openshift.io/type: build
+  name: collector-on-pull-request
+  namespace: rh-acs-tenant
+
+spec:
+
+  params:
+  - name: dockerfile
+    value: collector/container/konflux.Dockerfile
+  - name: git-url
+    value: '{{repo_url}}'
+  - name: image-expires-after
+    value: '13w'
+  - name: output-image
+    value: quay.io/redhat-user-workloads/rh-acs-tenant/acs/collector:on-pr-{{revision}}
+  - name: path-context
+    value: .
+  - name: revision
+    value: '{{revision}}'
+  - name: rebuild
+    value: 'true'
+  # TODO(ROX-20234): Enable hermetic builds
+  # - name: hermetic
+  #   value: "true"
+  # No language dependencies are required for central-db image.
```

Comment:
[nit] It's not `central-db`.
I believe, collector relies on git submodules. Do we get them checked out as part of `clone-repository` without any extra configuration on our side?

Reply:
We do check out the submodules by default: https://github.com/redhat-appstudio/build-definitions/blob/main/task/git-clone/0.1/git-clone.yaml#L28-L31

Reply:
This makes sense, thanks!

I'm a bit torn though if we should provide explicit `submodules`=`true` in `clone-repository` params
- to make it visible for readers that we request checking out submodules,
- to prevent from a case when the default `submodules` value suddenly becomes `false` and our builds break,
- to catch early a case when `submodules` gets renamed which may come with other behavior change.

WDYT?

Reply:
1) visibility -> agree
2+3) new parameter or default values would require a new task version according to https://github.com/redhat-appstudio/build-definitions/?tab=readme-ov-file#versioning, with the automated bot linking to the migration notes.

Specifying the parameter would only help for visibility. Let's try it.